drop-boxes/register-omero-metadata/register-omero.py

```diff
@@ -325,39 +344,85 @@ def process(transaction):
+		# The string "./" is set to point to the relative root folder
+		relativeFolderPath = getImageFolderPathFromLine(line)
+		if relativeFolderPath == "./":
+			relativeFolderPath = ""
+
+		log_print("RelativeFolderPath: " + relativeFolderPath)
 		# Due to the datahandler we need to add another subfolder of the same name to the path
-		imageFolder = os.path.join(incomingPath, folderName)
-		imageFile = os.path.join(imageFolder, fileName)
-		print "New incoming image file for OMERO registration:\t" + imageFile
+		basePath = os.path.join(incomingPath, folderName)
+		log_print("BasePath: " + basePath)
+		imageFolderPath = os.path.join(basePath, relativeFolderPath)
+		log_print("Incoming folder path for OMERO import: " + imageFolderPath)
+
+		# Look for overriding SAMPLE_ID in line/property map
+		if "SAMPLE_ID" in properties.keys():
+			if len(properties["SAMPLE_ID"]) == 10:
+				line_project_code = properties["SAMPLE_ID"][:5]
+				line_sample_code = properties["SAMPLE_ID"]
+				omero_dataset_id = registrationProcess.requestOmeroDatasetId(project_code=line_project_code, sample_code=line_sample_code)
+			else:
+				omero_dataset_id = default_omero_dataset_id
+		else:
+			omero_dataset_id = default_omero_dataset_id
+		log_print("Iteration omero dataset id: " + str(omero_dataset_id))
+
 
 		# 4. After we have received the omero dataset id, we know where to attach the image to
 		# in OMERO. We pass the omero dataset id and trigger the image registration process in OMERO.
-		omero_image_ids = registrationProcess.registerImageFileInOmero(imageFile, omero_dataset_id)
-		print "Created OMERO image identifiers:\t" + str(omero_image_ids)
+		# omero_image_ids = registrationProcess.registerImageFileInOmero(imageFile, omero_dataset_id)
+		omero_image_ids = registrationProcess.registerImageFolder(imageFolderPath, omero_dataset_id)
+		
+		log_print("Created OMERO image identifiers: " + str(omero_image_ids))
 
 		omero_failed = len(omero_image_ids) < 1
 		if omero_failed:
 			raise ValueError("Omero did not return expected image ids.")
 
-		# 5. Additional metadata is provided in an own metadata TSV file. 
-		# We extract the metadata from this file.
-		properties = getPropertyMap(line, property_names)
-
 		# 5.1 Validate metadata for image file
-		ImagingMetadataValidator.validateImagingProperties(getValidationMap(properties))
+		# ImagingMetadataValidator.validateImagingProperties(getValidationMap(properties))
 		
-		#one file can have many images, iterate over all img ids
+		# Annotate with metadata, using OMERO key-value pairs and tags 
+		# one file can have many images, iterate over all img ids
 		for img_id in omero_image_ids:
 			registrationProcess.registerOmeroKeyValuePairs(img_id, filterOmeroPropertyMap(properties, PROPPERTY_FILTER_LIST))
 
+		if "OMERO_TAGS" in properties.keys():
+			tag_list = properties["OMERO_TAGS"].split(",")
+			for tag_value in tag_list:
+				for img_id in omero_image_ids:
+					registrationProcess.tagOmeroImage(img_id, tag_value)
```

Comment:
What happens here if OMERO_TAGS is not in properties.keys()?

Reply:
If there was no OMERO_TAG col. in the tsv file, then don't tag the images. The tags should be a string of tag separated by commas, I should comment it in the code and readme doc.

Reply:
That would be perfect thank you 🙏